### PR TITLE
boost: don't require b2 when creating header-only package

### LINKED ISSUE #3028: [package] boost/1.73.0: header-only version has (unnecessary) dependency to b2
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **boost/1.73.0**
  * Operating System+version: **Linux Ubuntu 20.04**
  * Compiler+version: **GCC 10**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
Put your profile output here
```

### Steps to reproduce (Include if Applicable)

Build boost with the following command line:

    conan create -s compiler.version=10 -o boost:header_only=True -o boost:error_code_header_only=True . boost/1.73.0@fschoenm/testing

This should create a header-only package of boost but still has a build dependency to b2. Commenting out that dependency still creates the package so it seems to be not needed.

### PR DESCRIPTION
**Pull request by fschoenm**
Specify library name and version:  **boost/***

Remove a build dependency when only creating header-only package. Fixes #3028.

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — APPROVED** [review]
@fschoenm Thanks for your contribution. Let's wait for EAP approval on the next week and 2 more reviewers before merging.

**Review by madebr — APPROVED** [review]

**Review by intelligide — DISMISSED** [review]

**Comment by fschoenm** [comment]
@uilianries  According to that ticket I have the EAP access now. But the status of my pull requests hasn't changed. Is there something else I need to do?

**Comment by uilianries** [comment]
@fschoenm Wait for next approval wave. Usually we approve every week. The mechanism works not only due our capacity in terms of CI, but also to avoid spam and bad usage of the CI

**Comment by intelligide** [comment]
@fschoenm Close and re-open this PR after 1 minute ;)

**Comment by uilianries** [comment]
@fschoenm Consider https://github.com/fschoenm/conan-center-index/pull/1

**Comment by fschoenm** [comment]
@uilianries Thanks, I can try that. But that change wasn't even related to my original pull request. I only removed a build dependency for header-only builds. How could that make the other build options fail?

**Review by uilianries — APPROVED** [review]

File: recipes/boost/all/conanfile.py
Comment (madebr) [inline recipes/boost/all/conanfile.py:144]: @uilianries shouldn't `self.settings.compiler.runtime` be converted to a string?
```suggestion
        if self.settings.compiler == "Visual Studio" and "MT" in str(self.settings.compiler.runtime) and self.options.shared:
```

Comment (uilianries) [reply recipes/boost/all/conanfile.py:144]: Yes, it prevents cases where runtime is None. Thanks @madebr

Comment (madebr) [reply recipes/boost/all/conanfile.py:144]: oh, I thought it was needed so it would match with `MT` and `MTd`.

Comment (uilianries) [reply recipes/boost/all/conanfile.py:144]: ah, not in this case, runtime is a SettingsItem, which is converted to str

**Review by madebr — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**
